### PR TITLE
bpo-45822: Respect PEP 263's coding cookies in the parser even if flags are not provided

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue45822

### CODE REVIEW

**Review by ammaraskar — APPROVED** [review]
Link to previous code for reference:

https://github.com/python/cpython/blob/fd206b680705de3e0408fd29c7943317563e61e1/Python/pythonrun.c#L1447

https://github.com/python/cpython/blob/fd206b680705de3e0408fd29c7943317563e61e1/Python/pythonrun.c#L156-L171

File: Modules/_testcapimodule.c [hunk 1]
```diff
@@ -381,6 +381,19 @@ static PyTypeObject _HashInheritanceTester_Type = {
...
+pycompilestring(PyObject* self, PyObject *obj) {
+    if (PyBytes_CheckExact(obj) == 0) {
+        PyErr_SetString(PyExc_ValueError, "Argument must be a bytes object");
+        return NULL;
+    }
+    const char *the_string = PyBytes_AsString(obj);
+    if (the_string == NULL) {
+        return NULL;
+    }
+    return Py_CompileString(the_string, "blech", Py_file_input);
```
Comment (Yhg1s) [inline Modules/_testcapimodule.c:394]:
```suggestion
    return Py_CompileString(the_string, "<string>", Py_file_input);
```

So it matches the test.

**Review by Yhg1s — APPROVED** [review]

File: Lib/test/test_capi.py [hunk 2]
```diff
@@ -1017,6 +1017,14 @@ def test_state_access(self):
                 with self.assertRaises(TypeError):
                     increment_count(1, 2, 3)
 
+    def test_Py_CompileString(self):
```
Comment (Yhg1s) [inline Lib/test/test_capi.py:1020]: Actually, while backporting this patch to our internal 3.9, I noticed this test method is in the wrong class. It should be in the CAPITest class.

Comment (pablogsal) [reply Lib/test/test_capi.py:1020]: I will submit a fix today

**Pull request merged**